services/snyk/snyk-vulnerability-npm.service.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const { NotFound } = require('../errors')
+const SynkVulnerabilityBase = require('./snyk-vulnerability-base')
+
+module.exports = class SnykVulnerabilityNpm extends SynkVulnerabilityBase {
+  static get route() {
+    return {
+      base: 'snyk/vulnerabilities/npm',
+      pattern: ':packageName(.+)',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Snyk Vulnerabilities for npm package',
+        pattern: ':packageName',
+        namedParams: {
+          packageName: 'mocha',
+        },
+        staticExample: this.render({ vulnerabilities: '0' }),
+      },
+      {
+        title: 'Snyk Vulnerabilities for npm package version',
+        pattern: ':packageName',
+        namedParams: {
+          packageName: 'mocha@4.0.0',
+        },
+        staticExample: this.render({ vulnerabilities: '1' }),
+      },
+      {
+        title: 'Snyk Vulnerabilities for npm scoped package',
+        pattern: ':packageName',
+        namedParams: {
+          packageName: '@babel/core',
+        },
+        staticExample: this.render({ vulnerabilities: '0' }),
+      },
+    ]
+  }
+
+  async handle({ packageName }) {
+    const url = `https://snyk.io/test/npm/${packageName}/badge.svg`
+
+    try {
+      const { vulnerabilities } = await this.fetch({
+        // Snyk returns an HTTP 200 with an HTML page when the specified
+        // npm package is not found.
+        url,
+      })
+      return this.constructor.render({ vulnerabilities })
+    } catch (e) {
```

Comment:
I'm guessing there is probably a better way of dealing with this, but I went with this for now. The problem is that if you ask Snyk about the vulnerabilities for an npm packge (or a version of a package) that doesn't exist, Snyk returns an HTTP 200 HTML page with text that includes a message about the package not existing, whereas Snyk will return a 404 if you ask about vulnerabilities for a GH repo that doesn't exist. 

Without this catch/throw, the badge was returning with a message from the BaseSvgScrapingService (`unparseable svg response`) that I felt would be confusing for end users.

Reply:
Agreed. I added a `404` error message to the `fetch` call, just in case Snyk updates their end to return a 404 for npm like they do for GH along with an inline comment. Are you thinking these existing comments should be modified/enhanced and/or that we should describe the API behavior elsewhere in the code?

Reply:
Ah, gotcha, I see it now. Thanks!